src/datasets/table.py

```diff
@@ -2090,6 +2087,8 @@ def cast_array_to_feature(array: pa.Array, feature: "FeatureType", allow_number_
             return pa.ListArray.from_arrays(array.offsets, _c(array_values, feature[0]))
         elif isinstance(feature, Sequence):
             if feature.length > -1:
+                if array.offset and feature.length * len(array) != len(array_values):
+                    array_values = array.values[array.offset * array.type.list_size :]
```

Comment:
We also need to specify where the slice ends, no?

```suggestion
                    array_values = array.values[array.offset * array.type.list_size : (array.offset + len(array)) * array.type.list_size]
```